src/main/java/bt/dapps/SignumArt3.java

```diff
@@ -0,0 +1,531 @@
+				if (UseSoulbound) {
+					unbound = false;
+				}
+				return;
+			}
+		}
+		if (status == STATUS_FOR_AUCTION) {
+			if (getBlockTimestamp().ge(auctionTimeout)) {
+				// auction timed out, apply the transfer if any
+				if (highestBidder != null) {
+					pay(); // pay the current owner
+					sendMessage(highestBidder.getId(), currentPrice,owner.getId(), trackNewOwner);
+					owner = highestBidder; // new owner
+					highestBidder = null;
+					status = STATUS_NOT_FOR_SALE;
+					auctionMaxPrice = ZERO;
+					cancelOfferIfPresent();
+					if (UseSoulbound) {
+						unbound = false;
+					}
+				}
+				// current transaction will be refunded below
+			} else if (getCurrentTxAmount() > currentPrice) {
+				// Conditions match, let's register the bid
+
+				// refund previous bidder, if some
+				if (highestBidder != null)
+					sendAmount(currentPrice, highestBidder);
+
+				highestBidder = getCurrentTxSender();
+				currentPrice = getCurrentTxAmount();
+				totalBidsReceived++;
+				sendMessage( highestBidder.getId(), currentPrice, trackNewAuctionBid);
+				
+				if(auctionMaxPrice > ZERO && currentPrice >= auctionMaxPrice) {
+					// max price reached, so we also end the auction
+					pay(); // pay the current owner
+					sendMessage(highestBidder.getId(), currentPrice,owner.getId(), trackNewOwner);
+					owner = highestBidder; // new owner
+					highestBidder = null;
+					status = STATUS_NOT_FOR_SALE;
+					auctionMaxPrice = ZERO;
+					cancelOfferIfPresent();
+					if (UseSoulbound) {
+						unbound = false;
+					}
+				}
+				return;
+			}
+		}
+		if (status == STATUS_FOR_BULKNFT) {
+			arguments = tx.getMessage();
+			seller = arguments.getValue1();
+			MaxSaleSize = getValue(INDEX_ONSALE,seller);
+			SalePrice = getValue(INDEX_PRICE_SELLER,seller);
+			Buying = getCurrentTxAmount()/SalePrice;
+			if (Buying > ZERO && Buying <= MaxSaleSize ){
+				//Remmove size from sale
+				saveValue(INDEX_ONSALE,seller,MaxSaleSize-Buying);
+				//Add Size to buyer
+				Quantity =  getValue(INDEX_OWNERS,getCurrentTxSender());
+				saveValue(INDEX_OWNERS,getCurrentTxSender(),Buying+Quantity);
+				//Execution of Buy
+				currentPrice =  getCurrentTxAmount();
+				amountToPlatform = currentPrice * platformFee / THOUSAND;
+				amountToRoyalties = currentPrice * royaltiesFee / THOUSAND;
+				totalPlatformFee += amountToPlatform;
+				totalRoyaltiesFee += amountToRoyalties;
+				totalTimesSold++;
+				sendAmount(amountToRoyalties, royaltiesOwner);
+				sendAmount(currentPrice - amountToPlatform - amountToRoyalties, seller);
+				// We cant send messages as trackerAccount is always the same :( 
+			}
+			return;
+
+		}
+
+		// send back funds of an invalid order
+		sendAmount(getCurrentTxAmount(), getCurrentTxSender());
+	}
+	
+	public void likeIt() {
+		//Checking with maps that one account can only make one like
+		// I assume that if nothing is set for the key-value it returns 0
+		if 	(getValues(INDEX_PRICE_SELLER,getCurrentTxSender())  == ZERO){
```

Comment:
`INDEX_PRICE_SELLER` should not be another index (and the assumption of returning zero if not present is correct)

Reply:
This is the index for the price of the number of nfts he/she has on sale